app/controllers/labels_controller.rb

```diff
@@ -2,12 +2,54 @@
 
 class LabelsController < ApplicationController
   load_and_authorize_resource
-  def search
-    results = Label.ransack(name_i_cont: params[:search]).result
-    paginated = results.paginate(per_page: 3, page: params[:page])
+
+  rescue_from CanCan::AccessDenied do |_exception|
+    redirect_to root_path, alert: t('controllers.authorization')
+  end
+
+  def index
+    @search = {}
+  end
+
+  def merge
+    label_ids = params[:label_ids]
+    new_name = params[:new_label_name]
+
+    if new_name.length.positive? && new_name.length < 15
```

Comment:
Since this is also being verified in the frontend, should we set the maximum label length as a constant in the label model and fill it into the frontend js page using embedded ruby?

Reply:
This would work and could be injected in the CoffeeScript file (remember to change the extension, e.g., make it `.js.coffee.erb`). As an alternative, you could also pass the number through a `data` attribute on an HTML node.